File: Java/Creature.java

class Creature
{
	private QPoint pos;
	private bool m_isCarnivore;
	private bool m_isAnimal;
	private int id;
	private QSize taille;

	private Creature(){
	}

	private virtual int aggressivite(void){
	}

	private virtual DNAGraph * statistiques(void){
	}

	private virtual bool canSee(const Creature *c){
	}

	private virtual bool canReproduce(void){
	}

	private virtual bool canReproduceWith(const Creature *c){
	}

	private virtual Creature * reproduceWith(const Creature *c){
	}

	private virtual void update(void){
	}

	private virtual bool isAlive(void){
	}

	private virtual QRect* position(void){
	}

	private virtual DNAChanger changeStats(void){
	}

	private virtual int health(void){
	}

	private virtual int hitPower(void){
	}

	private virtual int defence(void){
	}

	private virtual int getId(void){
	}

	private virtual void attack(Creature *c){
	}

	private virtual bool isCarnivore(void){
	}

	private virtual bool isAnimal(void){
	}
}

File: Creature.cpp
#include "Creature.h"

Creature::Creature(){

}

Creature::~Creature(){

}

int
Creature::aggressivite(void){
    return 0;
}

bool
Creature::canSee(const Creature *c){
    return true;
}

bool
Creature::canReproduce(void){
    return true;
}

bool
Creature::canReproduceWith(const Creature *c){
    return true;
}

Creature*
Creature::reproduceWith(const Creature *c){
    return NULL;
}

void
Creature::update(void){}

bool
Creature::isAlive(void){
    return true;
}

QRect*
Creature::position(void){
    return NULL;
}

int
Creature::health(void){
    return 0;
}

int
Creature::hitPower(void){
    return 0;
}

int
Creature::defence(void){
    return 0;
}

int
Creature::getId(void){
    return 0;
}

void
Creature::attack(Creature *c){}

bool
Creature::isCarnivore(void){
    return true;
}

bool
Creature::isAnimal(void){
    return true;
}
